DragaliaAPI/DragaliaAPI/Features/StorySkip/StorySkipService.cs

```diff
@@ -0,0 +1,191 @@
+using DragaliaAPI.Features.Reward;
+using DragaliaAPI.Models.Generated;
+using DragaliaAPI.Services;
+using DragaliaAPI.Services.Game;
+using DragaliaAPI.Shared.Definitions.Enums;
+using DragaliaAPI.Shared.Features.StorySkip;
+using DragaliaAPI.Shared.MasterAsset;
+using DragaliaAPI.Shared.MasterAsset.Models.QuestRewards;
+using static DragaliaAPI.Shared.Features.StorySkip.FortConfigurations;
+
+namespace DragaliaAPI.Features.StorySkip;
+
+public class StorySkipService(
+    IFortRepository fortRepository,
+    IFortService fortService,
+    ILogger<StorySkipService> logger,
+    IQuestCompletionService questCompletionService,
+    IQuestRepository questRepository,
+    IQuestTreasureService questTreasureService,
+    IRewardService rewardService,
+    IStoryService storyService,
+    IUpdateDataService updateDataService,
+    IUserDataRepository userDataRepository
+) : IStorySkipService
+{
+    private async Task<QuestMissionStatus> CompleteQuestMissions(int questId, bool[] currentState)
+    {
+        List<AtgenMissionsClearSet> clearSet = new();
+
+        bool[] newState = { true, true, true };
+
+        QuestRewardData rewardData = MasterAsset.QuestRewardData[questId];
+        for (int i = 0; i < 3; i++)
+        {
+            (QuestCompleteType type, int value) = rewardData.Missions[i];
+
+            if (!currentState[i])
+            {
+                newState[i] = true;
+                (EntityTypes entity, int id, int quantity) = rewardData.Entities[i];
+                await rewardService.GrantReward(new Entity(entity, id, quantity));
+                logger.LogDebug("Completed quest mission {missionId}", i);
+                clearSet.Add(new AtgenMissionsClearSet(id, entity, quantity, i + 1));
+            }
+        }
+
+        List<AtgenFirstClearSet> completeSet = new();
+
+        if (currentState.Any(x => !x) && newState.All(x => x))
+        {
+            await rewardService.GrantReward(
+                new Entity(
+                    rewardData.MissionCompleteEntityType,
+                    rewardData.MissionCompleteEntityId,
+                    rewardData.MissionCompleteEntityQuantity
+                )
+            );
+            logger.LogDebug("Granting bonus for completing all missions");
+            completeSet.Add(
+                new AtgenFirstClearSet(
+                    rewardData.MissionCompleteEntityId,
+                    rewardData.MissionCompleteEntityType,
+                    rewardData.MissionCompleteEntityQuantity
+                )
+            );
+        }
+
+        return new QuestMissionStatus(newState, clearSet, completeSet);
+    }
+
+    public async Task IncreaseFortLevels()
+    {
+        Dictionary<FortPlants, FortConfig> fortConfigs = FortConfigurations.FortConfigs;
+
+        foreach (KeyValuePair<FortPlants, FortConfig> keyValuePair in fortConfigs)
+        {
+            FortPlants fortPlant = keyValuePair.Key;
+            FortConfig fortConfig = keyValuePair.Value;
```

Comment:
Nitpick: you can destructure here to make this cleaner

```suggestion
        foreach ((FortPlants fortPlant, FortConfig fortConfig) in fortConfigs)
        {
```

Reply:
This is good to know! I'll try it out.